### PR TITLE
[000] setup repo

### PR DESCRIPTION
**Pull request by TaylorJCrawford**

### CODE REVIEW

File: .DS_Store
Comment (glenn-hamilton) [inline .DS_Store:1]: Can you remove this file

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,40 @@
+db.properties
```
Comment (glenn-hamilton) [inline .gitignore]: You aren't going to be using `db.properties` so this can be removed

File: target/DropwizardTheITCrowd-1.0-SNAPSHOT-sources.jar
Comment (glenn-hamilton) [inline target/DropwizardTheITCrowd-1.0-SNAPSHOT-sources.jar:1]: The `target` folder needs removed from the commit

File: src/main/java/org/kainos/ea/DropwizardTheITCrowdServiceConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package org.kainos.ea;
+
+import io.dropwizard.Configuration;
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/DropwizardTheITCrowdServiceConfiguration.java:3]: This is the only import that is used here, the reset can be removed

File: src/main/java/org/kainos/ea/api/ConnectionService.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.kainos.ea.api;
+
+import org.kainos.ea.client.DatabaseConnectionFailedException;
+import org.kainos.ea.db.ConnectionDAO;
+
+public class ConnectionService {
+
+    private ConnectionDAO connectionDAO = new ConnectionDAO();
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/api/ConnectionService.java]: This can be final

File: src/main/java/org/kainos/ea/db/ConnectionDAO.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package org.kainos.ea.db;
+
+import org.kainos.ea.client.DatabaseConnectionFailedException;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/db/ConnectionDAO.java]: This import isn't used

File: src/main/java/org/kainos/ea/db/ConnectionDAO.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+public class ConnectionDAO {
+
+    private DatabaseConnector databaseConnector = new DatabaseConnector();
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/db/ConnectionDAO.java]: This can be final

File: src/main/java/org/kainos/ea/db/ConnectionDAO.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+
+    public String testDatabaseConnection() throws DatabaseConnectionFailedException {
+
+        try (Connection c = databaseConnector.getConnection()) {
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/db/ConnectionDAO.java]: `getConnection` is static, but you are calling it in a way that means `getConnection` doesn't need to be static

File: src/main/java/org/kainos/ea/db/ConnectionDAO.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            Statement st = c.createStatement();
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/db/ConnectionDAO.java]: This can go in the `try-with-resource`

Comment (TaylorJCrawford) [reply src/main/java/org/kainos/ea/db/ConnectionDAO.java]: Done

File: src/main/java/org/kainos/ea/db/ConnectionDAO.java (hunk 6 above)
Comment (TaylorJCrawford) [reply src/main/java/org/kainos/ea/db/ConnectionDAO.java]: Done

File: src/main/java/org/kainos/ea/db/ConnectionDAO.java (hunk 5 above)
Comment (TaylorJCrawford) [reply src/main/java/org/kainos/ea/db/ConnectionDAO.java]: Done

File: src/main/java/org/kainos/ea/db/ConnectionDAO.java (hunk 4 above)
Comment (TaylorJCrawford) [reply src/main/java/org/kainos/ea/db/ConnectionDAO.java]: Done

File: src/main/java/org/kainos/ea/api/ConnectionService.java (hunk 3 above)
Comment (TaylorJCrawford) [reply src/main/java/org/kainos/ea/api/ConnectionService.java]: Done

File: src/main/java/org/kainos/ea/DropwizardTheITCrowdServiceConfiguration.java (hunk 2 above)
Comment (TaylorJCrawford) [reply src/main/java/org/kainos/ea/DropwizardTheITCrowdServiceConfiguration.java:3]: Done

File: target/DropwizardTheITCrowd-1.0-SNAPSHOT-sources.jar
Comment (TaylorJCrawford) [reply target/DropwizardTheITCrowd-1.0-SNAPSHOT-sources.jar:1]: Done

File: .gitignore (hunk 1 above)
Comment (TaylorJCrawford) [reply .gitignore]: Done

File: .DS_Store
Comment (TaylorJCrawford) [reply .DS_Store:1]: Done

**Review by miagreen23 — APPROVED** [review]
Looks good from what I can see!

File: src/main/java/org/kainos/ea/db/ConnectionDAO.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+
+    public String testDatabaseConnection() throws DatabaseConnectionFailedException {
+
+        try (Connection c = databaseConnector.getConnection(); Statement st = c.createStatement()) {
+
+            ResultSet rs = st.executeQuery(
+                    "SHOW STATUS WHERE Variable_name='Uptime'");
+
+            while (rs.next()) {
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/db/ConnectionDAO.java]: This can be changed to ` if (rs.next())`

Comment (TaylorJCrawford) [reply src/main/java/org/kainos/ea/db/ConnectionDAO.java]: Done

**Review by glenn-hamilton — APPROVED** [review]

**Pull request merged**